pagination/header.go

```diff
@@ -0,0 +1,49 @@
+package pagination
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+func header(u *url.URL, rel string, limit, offset int) string {
+	q := u.Query()
+	q.Set("limit", strconv.Itoa(limit))
+	q.Set("offset", strconv.Itoa(offset))
+	u.RawQuery = q.Encode()
+	return fmt.Sprintf("<%s>; rel=\"%s\"", u.String(), rel)
+}
+
+func Header(u *url.URL, count int, limit, offset int) http.Header {
```

Comment:
I think it would make sense to change the signature a bit:

```go
func SetHeader(header http.Header, count int, limit, offset int)
```

Reply:
In order to build the links I need the URL that was requested.

The usage I envisioned kind of went like this:

```
func handler(r *http.Request, w http.ResponseWriter) {
    limit, offset := pagination.Parse(r, 100, 0, 500)

    // n = total
    c, n, err := h.r.ClientManager().GetClients(r.Context(), limit, offset)

    // ...

    pagination.Header(r.URL, n, limit, offset).Write(w)
    h.r.Writer().Write(w, r, c)
}
```

Reply:
Oh I didn't know that works! That's great!